### DISCUSSION TITLE
How do I get a step to appear on a dialog?

### DISCUSSION THREAD

**Discussion by iamcootis** [hakimio/ngx-ui-tour#187] [Q&A] [answered]
I have a step where a user needs to click a button that opens a dialog. The 2nd step anchor is on the dialog, but it never appears. Is there a way to make this happen?

On further inspection, it appears the tourAnchor inside the dialog is not being registered. I attempted to stop the previous tour and initialize and start a new tour on the dialog. This didn't work either. The anchor is still not registered.

**Comment by hakimio** [comment]
This should work if you mark your step as `async` (`isAsync` step config).

**Reply by iamcootis** [reply]
It was marked as isAsync. Still it never appears or show that the anchor is registered. BTW, I am using Angular 18.1.3

**Reply by hakimio** [reply, marked as answer]
Did you import `TourMatMenuModule` in your dialog component? Can you recreate the issue on [StackBlitz](https://stackblitz.com/edit/angular-wau2z8?file=src%2Fapp%2Fui-tour-demo.component.ts) or in a new github repo?

**Reply by iamcootis** [reply]
This appears to be the issue.
